scalding-core/src/test/scala/com/twitter/scalding/typed/gen/TypedPipeGen.scala

```diff
@@ -0,0 +1,260 @@
+package com.twitter.scalding.typed.gen
+
+import com.twitter.scalding.TypedPipe
+import org.scalacheck.{Cogen, Gen}
+
+object TypedPipeGen {
+  def pipeOfStd: Gen[(TypeWith[TypeGen], TypedPipe[TypeWith[TypeGen]#Type])] =
+    TypeGen.std.flatMap { t =>
+      pipeOf(t).map(t -> _)
+    }
+
+  def pipeOf(a: TypeWith[TypeGen])(implicit tg: Gen[TypeWith[TypeGen]]): Gen[TypedPipe[a.Type]] =
```

Comment:
Do you find that this sometimes blows up and generates gigantic graphs?

I often find I need to put a max depth or max size parameter on these kinds of recursive generators. But if this is working for you then it's definitely a lot cleaner to do it this way.

Reply:
Yeh, at first run it always blow up with StackOverFlow, with `frequency` and bigger weight on terminated leaf works now, but we can do it smarter if you like to propose.